lib/galaxy/jobs/command_factory.py

```diff
@@ -119,6 +124,7 @@ def build_command(
         # xref https://github.com/galaxyproject/galaxy/issues/3289
         commands_builder.prepend_command(PREPARE_DIRS)
 
+    for_pulsar = "script_directory" in remote_command_params
```

Comment:
Seems a bit confusing that `for_pulsar` is re-defined here, can the 2 "in" tests give different outcomes?

Reply:
```suggestion
```